src/suites/cts/validation_test.ts

```diff
@@ -0,0 +1,233 @@
+import GLSL from '../../tools/glsl.macro.js';
+
+import { GPUTest } from './gpu_test.js';
+
+export class ValidationTest extends GPUTest {
+  uniformBuffer: GPUBuffer = undefined!;
+  storageBuffer: GPUBuffer = undefined!;
+  sampler: GPUSampler = undefined!;
+  sampledTexture: GPUTexture = undefined!;
+
+  async init(): Promise<void> {
+    await super.init();
+
+    this.uniformBuffer = this.device.createBuffer({
+      size: 1024,
+      usage: GPUBufferUsage.UNIFORM,
+    });
+
+    this.storageBuffer = this.device.createBuffer({
+      size: 1024,
+      usage: GPUBufferUsage.STORAGE,
+    });
+
+    this.sampler = this.device.createSampler();
+
+    this.sampledTexture = this.device.createTexture({
+      size: { width: 16, height: 16, depth: 1 },
+      format: 'rgba8unorm',
+      usage: GPUTextureUsage.SAMPLED,
+    });
+  }
+
+  getErrorBuffer(): GPUBuffer {
+    return this.device.createBuffer({
+      size: 1024,
+      usage: 0xffff, // Invalid GPUBufferUsage
+    });
+  }
+
+  expectUncapturedError(commandBuffer: GPUCommandBuffer[] = []): Promise<void> {
```

Comment:
Our experience in Dawn is that is is very useful to know exactly when a validation error occurred. When updating to use `pushErrorScope` and `popErrorScope` I suggest making it so tests canbe written similar to this:

```
expectValidationError(() => {device.createBuffer(descriptor)});
```

A validation error must happen in the function passed to `expectValidationError` and no validation error must happen in the rest of the test (this can be checked by wrapping the whole test in an error scope).

Reply:
> this can be checked by wrapping the whole test in an error scope

(which we should do, for all tests)